error-issue/test/error_monitor.test.ts

```diff
@@ -0,0 +1,208 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ErrorMonitor} from '../src/error_monitor';
+import {StackdriverApi} from '../src/stackdriver_api';
+import {Stackdriver} from '../src/types';
+
+import nock from 'nock';
+import {mocked} from 'ts-jest/utils';
+
+describe('ErrorMonitor', () => {
+  let monitor: ErrorMonitor;
+  const stackdriver = ({
+    listGroups: jest.fn(),
+    setGroupIssue: jest.fn(),
+  } as unknown) as StackdriverApi;
+
+  const infrequentGroup: Stackdriver.ErrorGroupStats = {
+    group: {
+      name: 'Error: Infrequent error',
+      groupId: 'infrequent_id',
+    },
+    count: 2000,
+    timedCounts: [
+      {
+        count: 4,
+        startTime: new Date('Feb 25, 2020'),
+        endTime: new Date('Feb 26, 2020'),
+      },
+    ],
+    firstSeenTime: new Date('Feb 20, 2020'),
+    representative: {message: 'Error: Infrequent error'},
+  };
+  const acknowledgedGroup: Stackdriver.ErrorGroupStats = {
+    group: {
+      name: 'Error: Acknowledged error',
+      groupId: 'acknowledged_id',
+      trackingIssues: [{url: 'https://github.com/blah/blah'}],
+    },
+    count: 20000,
+    timedCounts: [
+      {
+        count: 6000,
+        startTime: new Date('Feb 25, 2020'),
+        endTime: new Date('Feb 26, 2020'),
+      },
+    ],
+    firstSeenTime: new Date('Feb 20, 2020'),
+    representative: {message: 'Error: Acknowledged error'},
+  };
+  const newGroup: Stackdriver.ErrorGroupStats = {
+    group: {
+      name: 'Error: New error',
+      groupId: 'new_id',
+    },
+    count: 20000,
+    timedCounts: [
+      {
+        count: 6000,
+        startTime: new Date('Feb 25, 2020'),
+        endTime: new Date('Feb 26, 2020'),
+      },
+    ],
+    firstSeenTime: new Date('Feb 20, 2020'),
+    representative: {message: 'Error: New error'},
+  };
+  const errorGroups = [infrequentGroup, acknowledgedGroup, newGroup];
+  const newReport = {
+    errorId: 'new_id',
+    firstSeen: new Date('Feb 20, 2020'),
+    dailyOccurrences: 6000,
+    stacktrace: 'Error: New error',
+  };
+
+  beforeAll(() => {
+    nock.disableNetConnect();
+  });
+
+  afterAll(() => {
+    nock.enableNetConnect();
+  });
+
+  beforeEach(() => {
+    monitor = new ErrorMonitor(stackdriver);
+    mocked(stackdriver.listGroups).mockResolvedValue(errorGroups);
+    nock.cleanAll();
+  });
+
+  afterEach(() => {
+    // Fail the test if there were unused nocks.
+    if (!nock.isDone()) {
+      throw new Error('Not all nock interceptors were used!');
```

Comment:
I don't remember if there's a more specific thing you can do it (like `self.fail(msg)` in Python tests)

You can also just do `nock.done()` which throws an error by itself and prints which ones have not been satisfied

Reply:
![image](https://user-images.githubusercontent.com/6694512/80115864-43883080-8553-11ea-83f8-4434798bad7b.png)

![image](https://user-images.githubusercontent.com/6694512/80115964-60246880-8553-11ea-8062-7d049f1a10de.png)

Reply:
I literally lol'd. Also I can't promise I'll remember this next time, feel free to have this screenshot on hand 😅😅😅